[OpenMP] Replace use of target address space with <gpuintrin.h> local

jhuber6 opened the pull request:
Summary:
This definition is more portable since it defines the correct value for
the target. I got rid of the helper mostly because I think it's easy
enough to use now that it's a type and being explicit about what's
`undef` or `poison` is good.

shiltian reviewed (commented):
Is this PR a duplication of another one?

jhuber6 commented:
> Is this PR a duplication of another one?

No it just pulls out this part to make it easier to land in pieces.

jplehr commented:
I did not see issues on this PR in local testing. Will take a closer look later today.

jhuber6 committed:
[OpenMP] Replace use of target address space with <gpuintrin.h> local

Summary:
This definition is more portable since it defines the correct value for
the target. I got rid of the helper mostly because I think it's easy
enough to use now that it's a type and being explicit about what's
undef or poison is good.

jhuber6 commented:
@jplehr @ronlieb 

AMD diff https://gist.github.com/jhuber6/9cb8a81e4b96a41cd7202a029c1242e9
